src/App.tsx

```diff
@@ -1,57 +1,83 @@
-import React from 'react';
-import './App.scss';
+import React, {
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
+import debounce from 'lodash.debounce';
+
 import { peopleFromServer } from './data/people';
 
+import { Person } from './types/Person';
+import { DELAY_OF_FILTERING } from './helpers/vars';
+
+import { SearchTitle } from './components/SearchTitle';
+import { UserDropDown } from './components/UserDropDown';
+
+import './App.scss';
+
+const filterUsers = (
+  users: Person[],
+  query: string,
+) => {
+  let preparedUsers = [...users];
+
+  if (query) {
+    const queryToLower = query.toLowerCase().trim();
+
+    preparedUsers = preparedUsers.filter(({ name }) => (
+      name.toLowerCase().includes(queryToLower)
+    ));
+  }
+
+  return preparedUsers;
+};
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [searchField, setSearchField] = useState('');
+  const [hasFocusField, setHasFocusField] = useState(true);
+  const [appliedSearchField, setAppliedSearchField] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+
+  const preparedUsers = useMemo(() => {
+    return filterUsers(peopleFromServer, appliedSearchField);
+  }, [peopleFromServer, appliedSearchField]);
+
+  const applieSearchField = useCallback(
```

Comment:
Here and in other places - please make sure your usage of `useCallback` is motivated
When used incorrectly it increases complexity of the code and hurts performance

Also, probably a typo here

Check out this add-on:
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker